### ISSUE TITLE
Event formatting in REST API

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#1898]
## Problem definition

We provide API endpoints to query events. Right now the blockchain events look different from internal events. We should decide and implement a common format for both.

Example event query for a channel:

```json
HTTP/1.1 200 OK
Content-Length: 1968
Content-Type: application/json
Date: Fri, 20 Jul 2018 10:08:08 GMT
mimetype: application/json

[
    {
        "address": "0x7692624c0C43285D51c0DdDfb2A6388f5002762E",
        "args": {
            "channel_identifier": "0x733ec6eecea8fb9fc2cb2b31ef3ae28cb751beedf956de43cca6943a1155c011",
            "closing_participant": "0xfB398E621c15E2BC5Ae6A508D8D89AF1f88c93e8"
        },
        "blockHash": "0x350ab9f001dd70fb8d38fa0861c3f90b16b221b1645d4bcc0a3447817d073d54",
        "blockNumber": 3674268,
        "block_number": 3674268,
        "event": "ChannelClosed",
        "logIndex": 2,
        "transactionHash": "0x9551bb75aa64b604410d843a1ef67bc57282eda394f17f66c9048f673d2f381c",
        "transactionIndex": 3
    }
]
```

## Proposed solution

We can split the endpoints into different endpoints for blockchain events and different ones for raiden internal events.

### Task 1

So that would need modifying the current event endpoints and adding two more.

In the end we should have:

- /blockchain_events/tokens/:token_address
- /blockchain_events/payment_networks/:token_address/channels/
- /blockchain_events/payment_networks/:token_address/channels/:partner_address
- /raiden_events/tokens/:token_address
- /raiden_events/networks/:token_address/channels/
- /raiden_events/networks/:token_address/channels/:partner_address

### Task 2

These endpoints for the internal events should be used for debugging. So we should remove the check  `EVENTS_EXTERNALLY_VISIBLE` from [here](https://github.com/raiden-network/raiden/blob/751a77c4ebcab341ad48a5ec1c8fbb56f8847bc4/raiden/api/python.py#L662) and [here](https://github.com/raiden-network/raiden/blob/751a77c4ebcab341ad48a5ec1c8fbb56f8847bc4/raiden/api/python.py#L709).

### Task 3 [Separate issue]

And when we do that we will need to also create a new API endpoint to query the history of the channel that will be using the `EVENTS_EXTERNALLY_VISIBLE` as defined in this [issue](https://github.com/raiden-network/raiden/issues/1949)

### Task 4

Remove the block number from internal events

**Comment by hackaugusto** [comment]
@LefterisJP what about removing the blockchain events from this endpoint?

**Comment by LefterisJP** [comment]
@hackaugusto @palango @CosminNechifor 

So we have two choices.

### Maintain different endpoints for internal and blockchain events

We can split the endpoints into different endpoints for blockchain events and different ones for raiden internal events.

So that would need modifying the current event endpoints and adding two more.

In the end we should have:

-    ` ('/events/tokens/blockchain/<hexaddress:token_address>', TokenBlockchainEventsResource)`
-    ` ('/events/tokens/raiden/<hexaddress:token_address>', TokenRaidenEventsResource)`
-    ` ('/events/channels/blockchain/<hexaddress:token_address>', ChannelsBlockchainEventsResource)`
-    ` ('/events/channels/raiden/<hexaddress:token_address>', ChannelsRaidenEventsResource)`
-    ` ('/events/channels/blockchain/<hexaddress:token_address>'/<hexaddress:partner_address>, ChannelBlockchainEventsResource)`
-    ` ('/events/channels/raiden/<hexaddress:token_address>/<hexaddress:partner_address>', ChannelRaidenEventsResource)`
 
### Drop all blockchain events from events API

As the name suggest we can also just drop all blockchain events from the REST API and only return raiden internal events. The user can create their own filters in geth and query them manually.


### Suggestion

I favour the first solution but it is indeed a bit more work.

**Comment by palango** [comment]
For me the question is what use cases we see for these endpoints. I don't see many besides debugging.

I think we should move the internal events to an history endpoint for a given channel or token network, where one can see what transfers were sent or received.

`/api/(version)/channels/history/(token_address)/(partner_address)`

**Comment by LefterisJP** [comment]
> For me the question is what use cases we see for these endpoints. I don't see many besides debugging.

For which endpoints? Internal raiden events or blockchain events?

**Comment by palango** [comment]
> For which endpoints? Internal raiden events or blockchain events?

I was talking about the current `/events` endpoint

**Comment by palango** [comment]
Fixed by #1982, removing the block number is tracked in #2170
